src/pages/index.js

```diff
@@ -75,30 +88,155 @@ addNewCardButton.addEventListener("click", () => {
   cardModal.open();
 });
 
+profileAvatarButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+
 //FUNCTIONS:
 
 function createCard(cardData) {
-  const newCard = new Card(cardData, "#card-template", () => {
-    imagePreviewModal.open(cardData);
-  });
+  const newCard = new Card(
+    cardData, //data
+    "#card-template", //cardSelector
+    () => {
+      //handleImageClick
+      imagePreviewModal.open(cardData);
+    },
+    handleDeleteSubmit,
+    handleLikeClick
+  );
+
   return newCard.generateCard();
 }
 
 function handleProfileFormSubmit(inputValues) {
-  console.log(inputValues.name);
-  userInfo.setUserInfo({
-    nameInput: inputValues.name,
-    jobInput: inputValues.job,
-  });
-  profileModal.close();
+  //call method that updates button text
+  profileModal.showButtonLoading("Saving...");
+  api
+    .updateUserInfo(inputValues.name, inputValues.job)
+    .then((res) => {
+      //call method to change back to default here
+      userInfo.setUserInfo({
+        nameInput: inputValues.name,
+        jobInput: inputValues.job,
+      });
+      profileModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileModal.hideButtonLoading("Save");
+    });
 }
 
 function handleAddCardFormSubmit(inputValues) {
   const name = inputValues.title;
   const link = inputValues.url;
-  const cardData = { name, link };
+  cardModal.showButtonLoading("Creating...");
+  api
+    .addNewCard(name, link)
+    .then((res) => {
+      cardsSection.prependItem(createCard(res));
+      cardModal.close();
+      cardModal.modalForm.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModal.hideButtonLoading("Create");
+    });
+}
+
+function handleEditAvatarFormSubmit(inputValues) {
+  const link = inputValues.url;
+  avatarModal.showButtonLoading("Saving...");
+  api
+    .updateAvatar(link)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      editAvatarValidator._disableButton();
```

Comment:
`_disableButton`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.



**Please, correct it everywhere**